### PR TITLE
Remove Python 3.7

### LINKED ISSUE #33708: Remove `Python 3.7` from the build matrix
With the addition of `3.12` and the deprecation of `3.7` we need our build pipelines to reflect the new reality.

- [ ] Remove `3.7` from any platform-matrixes within the `azure-sdk-for-python` repository
- [ ] Confirm that `analyze` checks now uses `3.8` instead of `3.7` (our previous lowest platform)
- [ ] Confirm the non-presence of `3.7` being used anywhere else in the build definitions

### PR DESCRIPTION
**Pull request by scbedd**
Resolves #33708 

@xiangyan99 let's see what exciting errors come out of this 😄 

Please note, I have included changes from #33729 , but that other PR should merge separately _before_ this one. I want to get signoff from mgmt team.

### CODE REVIEW

File: eng/ci_tools.txt [hunk 1]
```diff
@@ -24,10 +24,8 @@ coverage==7.2.5
 Jinja2==3.1.2
 MarkupSafe==2.1.3
 json-delta==2.0
-readme_renderer==37.3; python_version <= '3.7'
 readme_renderer==42.0; python_version > '3.7'
```
Comment (xiangyan99) [inline eng/ci_tools.txt:27]: We can remove python_version > '3.7' too? :)

Comment (scbedd) [reply eng/ci_tools.txt:27]: Sure. I'll do it.

File: tools/azure-sdk-tools/setup.py [hunk 2]
```diff
@@ -60,6 +58,7 @@
         ":python_version<'3.11'": ["tomli==2.0.1"],
         "build": ["six", "setuptools", "pyparsing", "certifi", "cibuildwheel"],
         "conda": ["beautifulsoup4"],
-        "systemperf": ["aiohttp>=3.0", "requests>=2.0", "tornado==6.0.3", "httpx>=0.21", "azure-core"]
+        "systemperf": ["aiohttp>=3.0", "requests>=2.0", "tornado==6.0.3", "httpx>=0.21", "azure-core"],
+         "ghtools": ["PyGithub>=1.59.0"],
```
Comment (scbedd) [inline tools/azure-sdk-tools/setup.py:62]:
```suggestion
        "ghtools": ["PyGithub>=1.59.0"],
```

**Review by xiangyan99 — APPROVED** [review]

**Comment by scbedd** [comment]
@xiangyan99 has confirmed that the failing identity build is a regression in the msal dev branch. We are g2g to merge.

**Pull request merged**